docs/public/specifications/tests/Nameserver-TP/nameserver15.md

```diff
@@ -0,0 +1,142 @@
+
+## Scope
+
+It is assumed that *Child Zone* is also tested and reported by [Connectivity01].
+This Test Case will just ignore non-responsive name servers or name servers not
+giving a correct DNS response for an authoritative name server.
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Summary
+
+Message Tag                 | Level   | Arguments                      | Message ID for message tag
+:---------------------------|:--------|:-------------------------------|:----------------------------------------------------------------------------------------------------------------------------
+N15_SOFTWARE_VERSION        | INFO    | ns_ip_list, query_name, string | The following name server(s) respond to version query "{query_name}" with string "{string}". Returned from name servers: "{ns_ip_list}"
+N15_NO_VERSION              | INFO    | ns_ip_list                     | The following name server(s) do not respond to version queries. Returned from name servers: "{ns_ip_list}"
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine Profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [Argument List].
+
+## Test procedure
+
+1. Create the following empty sets:
+   1. Name server IP, query name and string ("TXT Data")
+   2. Name server IP ("No Version")
+
+2. Create [DNS Query] with query type TXT and query class CHAOS ("TXT Query").
+
+3. Create the set of query names with values "version.bind"
+   and "version.server" ("Query Names").
+
+4. Obtain the set of name server IP addresses using [Method4] and
+   [Method5] ("Name Server IP").
+
+5. For each name server in *Name Server IP* do:
+   1. For each query name in *Query Names* do:
+      1. [Send] *TXT Query* with query name to
+         the name server and collect the response.
+      2. If at least one of the following criteria is met, go to next query name:
+         1. There is no DNS response.
+         2. The [DNS Response] does not have the [RCODE Name] NoError.
+         3. The [DNS Response] does not have any TXT record in the
+            answer section with query name as owner name.
+      3. For each TXT record in the answer section of the [DNS Response]
+         with owner name query name, extract and [concatenate] the string(s)
+         from the RDATA of the record.
+      4. If the extracted string is non-empty, add *Name Server IP*, query name
+         and the string to the *TXT Data* set.
+   2. If nothing was added to the *TXT Data* set in the previous steps,
+      add name server to the *No Version* set.
+
+6. If the *TXT Data* set is non-empty, then, for each unique
+   TXT string in the set, output *[N15_SOFTWARE_VERSION]* with
+   name server IP list, query name and TXT string.
```

Comment:
"...for each unique combination of string and query name in the set, output..."

In the definition of the set the field is "string" not "TXT string".